machine_learning/data_downloader.py

```diff
@@ -0,0 +1,160 @@
+import argparse
+import requests
+from urllib.parse import urlparse
+from web_monitoring.db import Client
+import pandas as pd
+import os
+import time
+from docopt import docopt
+import warnings
+
+def parse_url(url):
+    """
+    Parse Versionista change urls to get version ids
+    https://versionista.com/{site id}/{page id}/{to version id}:{from version id}
+
+    Parameters
+    ----------
+    url : string
+
+    Returns
+    -------
+    ids : dictionary
+        Site, page, and version ids of the from and two versions extracted from the url
+    """
+    try:
+        result = urlparse(url)
+        ids = [element for element in result.path.split('/') if element]
+        site_id = ids[0]
+        page_id = ids[1]
+        version_ids = ids[2].split(':')
+        to_version_id = version_ids[0]
+        from_version_id = version_ids[1]
+        ids = {'site_id': site_id, 'page_id': page_id,
+               'from_version_id': from_version_id, 'to_version_id': to_version_id}
+        return ids
+    except:
+        warnings.warn('Url "' + url + '" is not in the required format and no data will be downloaded for this row', Warning)
+
+def get_storage_uris(url):
+    """
+    Get uris of the versions stored in our db
+    Uses get_version_by_versionista_id from web_monitoring.db
+
+    Parameters
+    ----------
+    url : string
+
+    Returns
+    -------
+    from_version_uri, to_version_uri : string
+        To and from version uris to access content in db
+    ids : dictionary
+        Site, page, and version ids of the from and two versions extracted from the url
+    """
+    ids = parse_url(url)
+    from_version_id = ids['from_version_id']
+    to_version_id = ids['to_version_id']
+    print(ids)
+    client = Client.from_env()
+    """
+    Some versionista uris have the format https://versionista.com/site_id/page_id/to_version_id:0/
+    which automatically directs to the diff with the previous version.
+    However, versions in our db are saved with their true version ids and when the from_version_id is '0',
+    this is handled by taking advantage of the optional parameters of get_version_uri.
+    """
+    if from_version_id == '0':
+        get_to_version = client.get_version_by_versionista_id(versionista_id=to_version_id)
+        to_version_uri = get_to_version['data']['uri']
+        valid_from_version_id = get_to_version['data']['source_metadata']['diff_with_previous_url'].split(':')[-1]
+        from_version_uri = client.get_version_by_versionista_id(versionista_id=valid_from_version_id)['data']['uri']
+        ids['from_version_id'] = valid_from_version_id
+    else :
+        to_version_uri = client.get_version_by_versionista_id(versionista_id=to_version_id)['data']['uri']
+        from_version_uri = client.get_version_uri(versionista_id=from_version_id)['data']['uri']
+    print(from_version_uri, to_version_uri)
```

Comment:
This debugging print should be removed. Library code shouldn't print, as a general rule, because callers using it have no way to suppress that printing.